src/guides/anatomy-of-an-http-transaction.md

```diff
@@ -0,0 +1,429 @@
+data.
+
+If you want, you can *explicitly* write the headers to the response stream.
+(Remember, the `response` object is a [`WritableStream`][].) To do this,
+there's a method called [`writeHead`][], which writes the status code and the
+headers to the stream.
+
+```javascript
+response.writeHead(200, {
+  'Content-Type': 'application/json',
+  'X-Powered-By': 'bacon'
+});
+```
+
+Once you've set the headers (either implicitly or explicitly), you're ready to
+start sending response data.
+
+## Sending Response Body
+
+Since the `response` object is a [`WritableStream`][], writing a response body
+out to the client is just a matter of using the usual stream methods.
+
+```javascript
+response.write('<html>');
+response.write('<body>');
+response.write('<h1>Hello, World!</h1>');
+response.write('</body>');
+response.write('</html>');
+response.end();
+```
+
+Remember, the `end` function on streams can also take in some optional data to
+send as the last bit of data on the stream, so we can simplify the example above
+as follows.
+
+```javascript
+response.end('<html><body><h1>Hello, World!</h1></body></html>');
+```
+
+> **Note:** It's important to set the status and headers *before* you start
+writing chunks of data to the body. This makes sense, since headers come before
+the body in HTTP responses.
+
+## Another Quick Thing About Errors
+
+The `response` stream can also emit `'error'` events, and at some point you're
+going to have to deal with that as well. All of the advice or `request` stream
+errors still applies here.
+
+## Put It All Together
+
+Now that we've learned about making HTTP responses, let's put it all together,
+building on the earlier example. We're going to make a contrived example where
+we send back all the useful data that was send to us by the user, in a JSON
+object, which we'll convert to a string using `JSON.stringify`.
+
+```javascript
+
+var http = require('http');
+
+http.createServer(function(request, response) {
+  var headers = request.headers;
+  var method = request.method;
+  var url = request.url;
+  var body = '';
+  request.on('error', function(err) {
+    console.error(err);
+  }).on('data', function(chunk) {
+    body += chunk;
+  }).on('end', function() {
+    // BEGINNING OF NEW STUFF
+
+    response.on('error', function(err) {
+      console.error(err);
+    });
+
+    response.statusCode = 200;
+    response.setHeader('Content-Type', 'application/json');
+    // Note: the 2 lines above could be replaced with this next one:
+    // response.writeHead(200, {'Content-Type': 'application/json'})
+
+    var responseBody = {
+      headers: headers,
+      method: method,
+      url: url,
+      body: body
+    };
+
+    response.write(JSON.stringify(responseBody));
+    response.end();
+    // Note: the 2 lines above could be replaced with this next one:
+    // response.end(JSON.stringify(responseBody))
+
+    // END OF NEW STUFF
+  });
+}).listen(8080);
+```
+
+## Echo Example
```

Comment:
I think it strikes the right balance of difference and similarity to the previously built-up example. I'll reword it a bit in order to call out both the difference and the similarity.